### PR TITLE
Small fixes for release

### PR DESCRIPTION
**Pull request by dsikka**
Summary
- Seems like trainer ends up on different device when running finetune scenarios with multiple gpus
- Limit to single gpu for now

### CODE REVIEW

File: tests/llmcompressor/transformers/finetune/test_finetune_no_recipe_custom_dataset.py [hunk 1]
```diff
@@ -122,7 +123,14 @@ class TestOneshotCustomDatasetSmall(TestFinetuneNoRecipeCustomDataset):
...
         import torch
 
-        self.device = "cuda:0" if torch.cuda.is_available() else "cpu"
+        self.monkeypatch = pytest.MonkeyPatch()
+
+        if torch.cuda.is_available():
+            self.device = "cuda:0"
+            self.monkeypatch.setenv("CUDA_VISIBLE_DEVICES", "0")
+        else:
+            self.device = "cuda:0"
```
Comment (horheynm) [inline tests/llmcompressor/transformers/finetune/test_finetune_no_recipe_custom_dataset.py:132]: if cuda is availble set to the first recognized - ok
else set to the first recognized? should this be cpu?

**Review by kylesayrs — APPROVED** [review]
Added to my list, definitely something we should look into at a later time. Might be worth adding a user warning about.

**Comment by dsikka** [comment]
> Added to my list, definitely something we should look into at a later time. Might be worth adding a user warning about.

Yeah I created a ticket for it. 
Seems like the loss functionality is angry at us

File: tests/llmcompressor/transformers/finetune/test_finetune_no_recipe_custom_dataset.py (hunk 1 above)
Comment (dsikka) [reply tests/llmcompressor/transformers/finetune/test_finetune_no_recipe_custom_dataset.py:132]: oops typo. good catch

File: tests/llmcompressor/transformers/finetune/test_finetune_no_recipe_custom_dataset.py [hunk 2]
```diff
@@ -122,7 +123,14 @@ class TestOneshotCustomDatasetSmall(TestFinetuneNoRecipeCustomDataset):
     def setUp(self):
         import torch
 
-        self.device = "cuda:0" if torch.cuda.is_available() else "cpu"
+        self.monkeypatch = pytest.MonkeyPatch()
+
+        if torch.cuda.is_available():
+            self.device = "cuda:0"
+            self.monkeypatch.setenv("CUDA_VISIBLE_DEVICES", "0")
```
[the remark this answers is not in the document]
Comment (dsikka) [reply tests/llmcompressor/transformers/finetune/test_finetune_no_recipe_custom_dataset.py:130]: let me check

File: tests/llmcompressor/transformers/finetune/test_finetune_no_recipe_custom_dataset.py (hunk 1 above)
Comment (rahul-tuli) [reply tests/llmcompressor/transformers/finetune/test_finetune_no_recipe_custom_dataset.py:132]: Yes, this should be cpu
```suggestion
            self.device = "cpu"
```

**Review by rahul-tuli — COMMENTED** [review]
Glad this worked!

**Review by rahul-tuli — APPROVED** [review]

**Review by horheynm — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]
